lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -8,11 +8,27 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
   def resolve_os_deps(native) do
     {cflags_list, libs_list} =
       native.os_deps
-      |> Enum.map(fn {provider_or_providers, lib_name_or_names} ->
-        providers = Bunch.listify(provider_or_providers)
-        lib_names = Bunch.listify(lib_name_or_names)
+      |> Enum.map(fn
+        {name, providers} when is_atom(name) and name != :pkg_config ->
+          resolve_os_dep(name, native.app, Bunch.listify(providers), [])
+
+        {providers, lib_names} ->
+          IO.warn("""
+          Native #{inspect(native.name)} uses deprecated syntax for `os_deps`. \
```

Comment:
It's out of the scope of this PR, but I have an impression that in multiple places we are using `native` term to describe `resources`, which can be either `lib` or a `native`. And I believe it also happens here.

Reply:
Yeah, this would be for a separate PR. Though I'd be for using the term `natives` to both 'regular natives' and libs, instead of `resources`